File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.cpp
class Solution {
public:
    int gcd(int a,int b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
    string gcdOfStrings(string str1, string str2) {
        string x = "";
        int len = 0;
        if (str1 + str2 != str2 + str1){
            return "";
        }else{
            len = gcd(str1.size(),str2.size());
        }
        return str1.substr(0,len);

    }
};

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
class Solution:
    def gcd(self, a, b):
        if b == 0:
            return a
        return self.gcd(b, a % b)

    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""

        length = self.gcd(len(str1), len(str2))
        return str1[:length]